spyderlib/widgets/file_switcher.py

```diff
@@ -0,0 +1,415 @@
+        """
+        count = self.tabs.count()
+        if count == 0:
+            self.accept()
+            return
+
+        if stack_index is not None:
+            # cache full paths, and short paths and invalidate line couts
+            self.full_index_to_path = [getattr(td, 'filename', None)
+                                       for td in self.tab_data]
+            current_path = self.original_path = \
+                self.full_index_to_path[stack_index]
+            self.full_index_to_short_path = \
+                shorten_paths(self.full_index_to_path)
+            self.path_to_line_count = None  # we only get this on demand
+        else:
+            current_path = self.current_path()  # could be None
+
+        # get filter text and optional line number
+        filter_text = to_text_string(self.edit.text()).lower()
+        trying_for_line_num = (':' in filter_text)
+        filter_text, line_num = filter_text.split(':', 1) \
+            if trying_for_line_num else (filter_text, "")
+        try:
+            line_num = int(line_num)
+        except ValueError:
+            line_num = -1
+
+        # cache line counts if we need now them
+        if trying_for_line_num and self.path_to_line_count is None:
+            self.path_to_line_count = \
+                {path: self.tabs.widget(idx).get_line_count()
+                 for idx, path in enumerate(self.full_index_to_path)}
+
+        self.listwidget.clear()
+        self.filtered_index_to_path = []
+        for index, path in enumerate(self.full_index_to_path):
+            text = to_text_string(self.tabs.tabText(index))
+            if len(filter_text) == 0 or filter_text in text.lower():
+                if len(filter_text) > 0:
+                    text = text.replace(filter_text,
+                                        '<u>' + filter_text + '</u>')
+                text = "<b>" + text + "</b>"
+                if trying_for_line_num:
+                    text += " [" + str(self.path_to_line_count[path]) + " " \
+                                    + _("lines") + "] "
+                text += "<br><span style='font-size:10px'>" \
+                    + self.full_index_to_short_path[index] + "</span>"
+                item = QListWidgetItem(self.tabs.tabIcon(index),
+                                       text, self.listwidget)
+                item.setToolTip(path)
+                item.setSizeHint(QSize(0, 25))
+                self.listwidget.addItem(item)
+                self.filtered_index_to_path.append(path)
+
+        if current_path is not None \
+                and current_path in self.filtered_index_to_path:
+            self.listwidget.setCurrentRow(self.filtered_index_to_path
+                                              .index(current_path))
+        elif len(self.filtered_index_to_path) > 0:
+            self.listwidget.setCurrentRow(0)
+        self.edit_line.emit(line_num)  # could be -1
```

Comment:
What do you mean with the comment ? That we could always put `-1` instead of `line_num` or that `line_num` can have the value `-1` ? And why is it important ?